src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -292,18 +313,21 @@ def create_connectedk8s(cmd, client, resource_group_name, cluster_name,
         except ApiException as e:
             print("Exception when calling CoreV1Api->list_namespaced_pod: %s\n" % e)
         for pod in api_response.items:
-            for container_status in pod.status.container_statuses:
-                if container_status.name == 'connect-agent':
-                    connect_agent_state = container_status.state.running
-                    if connect_agent_state is not None:
-                        found_running = found_running + 1
-                        time.sleep(2)
+            if pod.metadata.name.startswith('connect-agent'):
+                for container_status in pod.status.container_statuses:
```

Comment:
Is is possible to merge checking container status and container availability into one step ?